docs/source/transaction_family_specifications/supplychain_transaction_family.rst

```diff
@@ -0,0 +1,255 @@
+storing records of asset location and ownership.
+
+State
+=====
+
+This section describes in detail how SCTP objects are stored and addressed in the global state.
+
+SCTP stores the following data objects:
+
+Agent
+-----
+
+Agents represent owners or holders of goods in the system. All agents must be registered in the system before they can be used.
+
+.. code-block:: javascript
+
+    {
+        Name: "name", // The name of the agent
+        Type: "type", // Agent type, domain specific value
+        Url: // The web address of the agent as a fully qualified URL
+        OwnRecords: [], // list of all records(by address) this agent owns
+        HoldRecords: [], // list of all records(by address) this agent holds
+        OpenApplications: [], // list of all applications(by records address) this agent owns
+        AcceptedApplications: [] // list of all records this agent owns
+    }
+
+For "HoldRecords", a value 1 indicates there is an open application against the record, 0 means no open application.
+
+For "OwnRecords" the value 1 means Held and Owned, 0 means Owned and not held.
+
+For "OpenApplications" indicates the agent has an application open against a record and is waiting for a response.
+
+For "AcceptedApplications" indicates the agent has an application accepted against a record and can now create a child. The value is the sensor ID associated with the record.
+
+
+Record
+------
+A record represents a good or material that is being tracked in the system.
+
+.. code-block:: javascript
+
+    {
+        RecordInfo: {
+            Owner: agent_address,  // the owner of the item
+            Custodians: [],  // A list of custodians(identified by address) for the record.
+            CurrentHolder: agent_address,  // The agent in possession of the item, either the Owner or the current Custodian
+            Timestamp: timestamp,  // time stamp of record creation, seconds from the epoc
+            Sensor: 'sensor_id',  // identifier of sensor attached to item
+            Final: true | false,  // once set true the record cannot be updated
+            ApplicationFrom: agent_address,  //  The address of the applicant
+            ApplicationType: "Owner" | "Custodian",  // Which field is the application for.
+            ApplicationTerms: "text",  // human readable terms of the transfer application.
+            ApplicationStatus:  "Open" | "Rejected" | "Cancelled"
+        }
+    }
+
+
+Addressing
+----------
+
+When a setting is read or changed, it is accessed by addressing it using the following algorithm:
+
+The SCTP uses the Supplychain namespace computed as the first 6 hex characters of:
+.. code-block:: javascript
+
+    namespace = sha512('Supplychain')).hexdigest()[:6]
+
+Agent addresses are generated sha512(agent_pub_key), prepended with the Supplychain namespace.
+
+.. code-block:: javascript
+
+    agent_addr = namespace + sha512(agent_pub_key)).hexdigest()
```

Comment:
I believe the full length of a sha512 hexdigest is 128 characters. This will generate an invalid address.